src/Exceptions/BadRequestException.php

```diff
@@ -26,20 +28,36 @@ class BadRequestException extends RuntimeException
      *
      * @param RequestInterface       $http_request
      * @param ResponseInterface|null $http_response
-     * @param string                 $message
-     * @param int                    $code
+     * @param string|null            $message
+     * @param int|null               $code
      * @param Throwable|null         $previous
      */
     public function __construct(RequestInterface $http_request,
                                 ?ResponseInterface $http_response = null,
-                                string $message = 'Unsuccessful request',
-                                int $code = 0,
+                                ?string $message = null,
+                                ?int $code = null,
                                 ?Throwable $previous = null)
     {
         $this->http_request  = $http_request;
         $this->http_response = $http_response;
 
-        parent::__construct($message, $code, $previous);
+        $service_error_message = $http_response instanceof ResponseInterface
```

Comment:
Maybe it makes sense to combine this with the code on line 52 (`$http_code = $http_response instanceof ResponseInterface ...`). No need to check instance of `$http_responce` twice.

Something like this:
```php
$service_error_message = null;
$http_code = null;
if ($http_response instanceof ResponseInterface) {
    $service_error_message = $this->extractErrorMessageFromResponse($http_response);
    $http_code = $http_response->getStatusCode();
}
```

Reply:
Yay! Approved